### ISSUE TITLE
Be a superset of CSS

### ISSUE THREAD

**Issue by curvedmark** [curvedmark/roole#60]
After arguing with myself for a long time, I decide to make Roole a superset of CSS. I know it's ironic since I was arguing against the CSS syntax in #36.

But I didn't see a few useful use cases then that I see now:
- Using Roole merely as a CSS prefixer, in which case #16 should be implemented.
- Importing other CSS frameworks directly:
  
  ```
  @void {
      @import 'bootstrap.css'
  }
  
  #submit {
      @extend .button
  }
  ```
  
  Only the button code in bootstrap will be generated.
  (The rule of selector matching might need be changed in order to support this, e.g., `@extend .btn` should also match `.btn.active`)
- namespacing CSS frameworks:
  
  ```
  @module .bs {
      @import 'bootstrap.css'
  }
  ```
  
  And class selectors in `bootstrap.css` will be prefixed with `bs-`

**Comment by stef-rausch** [comment]
:+1:

**Comment by curvedmark** [comment]
To support the second use case, a construct like `@extend-all .button` might need to be introduced, which will match any selector that contains the specified selector.

For example, `@extend-all .button` matches `.button`, `.button.active` and `.button .icon`.

**Comment by curvedmark** [comment]
Issue regarding `@extend-all`: #63, and `@module`: #62

**Comment by curvedmark** [comment]
Landed in 76c0600f6473adfa21fd77562f7d7c81e4b69a19

**Comment by stef-rausch** [comment]
:+1:
